core/src/main/java/hivemall/smile/tools/MaxEntPredictUDF.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    @Override
+    public Object evaluate(@Nonnull DeferredObject[] arguments) throws HiveException {
+        Object arg0 = arguments[0].get();
+        if (arg0 == null) {
+            throw new HiveException("Model was null");
+        }
+        
+        Text model = modelOI.getPrimitiveWritableObject(arg0);
+        try {
+			evaluator = new SepDelimitedTextGISModelReader(model).constructModel();
+		} catch (IOException e) {
+			throw new HiveException(e.getMessage());
+		}
+        
+        Object arg1 = arguments[1].get();
+        if (arg1 == null) {
+            throw new HiveException("arguments were null");
+        }
+        attributes = SmileExtUtils.resolveAttributes(attributesOI.getPrimitiveWritableObject(arg1).toString());
+        
+        Object arg2 = arguments[2].get();
+        if (arg2 == null) {
+            throw new HiveException("array<double> features was null");
+        }
+        this.featuresProbe = parseFeatures(arg2, featuresProbe);
+
+        double[] obs = this.featuresProbe.toArray();
+        
+        String[] names = new String[obs.length];
+  	    float[] values = new float[obs.length];
+	    for (int i = 0; i < obs.length; i++){
+	    	  if (attributes[i].type == AttributeType.NOMINAL){
+	    		  names[i] = i + "_" + String.valueOf(obs[i]).toString();
+	    		  values[i] = Double.valueOf(1.0).floatValue();
+	    	  }else{
+	    		  names[i] = String.valueOf(i);
+	    		  values[i] = Double.valueOf(obs[i]).floatValue();
+	    	  }
+	    }
+        
+        double[] ocs = evaluator.eval(names,values);
+        String klass = evaluator.getBestOutcome(ocs);
+        
+        List<DoubleWritable> ocss = new ArrayList<DoubleWritable>();
+        for (int i = 0; i < ocs.length; i++){
+        	ocss.add(new DoubleWritable(ocs[i]));
+        }
+        
+        return new Object[]{new Text(klass), ocss};
+    }
+
+    @Nonnull
+    private Vector parseFeatures(@Nonnull final Object argObj, @Nonnull Vector probe) {
+    	final int length = featureListOI.getListLength(argObj);
+    	if (probe == null) {
+            probe = new DenseVector(length);
+        } else if (length != probe.size()) {
+            probe = new DenseVector(length);
+        }
+    	
+    	for (int i = 0; i < length; i++) {
+            Object o = featureListOI.getListElement(argObj, i);
+            if (o == null) {
+                continue;
+            }
+            double v = PrimitiveObjectInspectorUtils.getDouble(o, featureElemOI);
+            probe.set(i, v);
+        } 
+        return probe;
+    }
+ 
+    @Override
+    public void close() throws IOException {
+        this.modelOI = null;
+        this.featureElemOI = null;
+        this.featureListOI = null;
+        this.evaluator = null;
+    }
+
+    @Override
+    public String getDisplayString(String[] children) {
+        return "tree_predict(" + Arrays.toString(children) + ")";
```

Comment:
`tree_predict` is not correct. Better to be renamed to`matext_predict` or something.